### DISCUSSION TITLE
Should CanvasItem.`update()` be renamed to `request_redraw()` or  `redraw()`?

### DISCUSSION THREAD

**Discussion by Mickeon** [godotengine/godot-proposals#5154] [Polls]
In the [more recent rename tracker for Godot 4.0](https://github.com/godotengine/godot/issues/54161#issuecomment-962015533), it was brought up to rename **CanvasItem**.`update()` to `request_redraw()`. It's clear that this method's name **needs** to be urgently changed. In a broader context, especially when extending a **CanvasItem** **Script**, the word "_update_" is vague, and `request_redraw()` is exactly what the method has been described to do. It's outstandingly meaningful.

[Personally](https://github.com/godotengine/godot/issues/54161#issuecomment-1034903362), I'd prefer to call it `redraw`. It's short and simple like `update()`, while still being more meaningful, but it *may* be misleading, as the redraw is actually done inside `NOTIFICATION_DRAW`, that the method schedules to send when a new frame is to be rendered, but this can be [explained in the documentation](https://github.com/godotengine/godot/pull/64345).

But regardless,

**Comment by dalexeev** [comment]
The winner is [`queue_redraw()`](https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#class-canvasitem-method-queue-redraw).

* godotengine/godot#64377
